src/org/jetuml/gui/EditorFrame.java

```diff
@@ -730,4 +731,81 @@ private void removeGraphFrameFromTabbedPane(DiagramTab pTab)
 		tabs().remove(pTab);
 		showWelcomeTabIfNecessary();
 	}
+
+	/**
+	 * A callback for Notification objects called when they reached the end of their lifetime.
+	 * It removes them from the handler notification list.
+	 */
+	public final class CleanUpCallback
+	{
+

```

Comment:
To prevent having two timers (one in `ToastNotification`, triggering the end of the animation, and one in `EditorFrame`, removing the notification from the notification list), I kept the callback logic, but used a Runnable and a lambda expression which makes it simpler and more readable. 
Moreover, we can keep the segregated access logic, allowing the callback to remove **only** the notification it is called from.